src/firecracker/src/api_server_adapter.rs

```diff
@@ -55,6 +55,12 @@ impl ApiServerAdapter {
         vmm: Arc<Mutex<Vmm>>,
         event_manager: &mut EventManager,
     ) -> Result<(), ApiServerError> {
+        // This wraps ApiServerAdapter inside an Arc<Mutex<...>> when it is not
+        // Send + Sync because the Vmm object (to which we hold a reference to) is not
+        // Send + Sync. The EventManager expects Arc<Mutex<dyn MutEventSubscriber>>
+        // kinds of things though, so here we are. This is *NOT* a problem, since we
+        // do not share the object among threads.
+        #[allow(clippy::arc_with_non_send_sync)]
```

Comment:
This is quite big red flag to me. We rely on Rust types for safety/sanity reasons. Breaking it now (even if it is "safe") gives us an excuse to do it again in the future. I am sure there are alternative designs that do not require this workaround.

Reply:
> We rely on Rust types for safety/sanity reasons

Sure, this thing does not revert you statement. What exactly are you worried about?